docs/roadmap.md

```diff
@@ -0,0 +1,81 @@
+# Project status & roadmap
+
+## Overview
+
+The Asterius project has come a long way and some examples with complex
+dependencies already work. It's still less mature than GHCJS yet; see the next
+section for details.
+
+In general, it's hard to give ETA for "production readiness", since improvements
+are continuous, and we haven't collected enough use cases from seed users yet.
+For more insight into what comes next for this project, we list our quarterly
+roadmap here.
+
+Besides the goals in each quarter, we also do regular maintainence like
+dependency upgrades and bugfixes. We also work on related projects (mainly
+`haskell-binaryen` and `inline-js`) to ensure they are kept in sync and also
+useful to regular Haskell developers.
+
+## What may stop one from using Asterius right now
+
+- Lack of JavaScriptCore/Safari support, due to incomplete JavaScript `BigInt`
+  support at the moment.
+- Runtime bugs. The generated code comes with a complex hand-written runtime
+  which is still buggy at times. The situation is expected to improve once we're
+  able to work with an IR more high-level than Cmm and shave off the current
+  hand-written garbage collector; see the 2020 Q3 section for explanation.
+- GHCJS projects aren't supported out of the box. Major incompatabilities
+  include:
+  - Word sizes differ. Asterius is still 64-bit based at the moment.
+  - JSFFI syntax and semantics differ. Asterius uses `Promise`-based async JSFFI
+    and GHCJS uses callbacks.
+  - Cabal handles GHCJS and Asterius differently.
+- Lack of Nix support.
+- Lack of GHCi support. TH support is not 100% complete; certain TH API which
+  requires preserving state across splices (e.g. `getQ`/`putQ`) don't work yet.
+- Lack of profiling support for generated code.
+
+## Quarterly roadmap
+
+### 2020 Q3
+
+Work in 2020 Q3 is focused on:
+
+- Introducing C/C++ toolchain support. The first step is to introduce libc in
+  the generated wasm code, and use libc functionality to replace certain runtime
+  functionality (e.g. memory management). Once we're confident our runtime and
+  generated code is compatible with libc, we'll look into building & linking C
+  source files in Haskell packages.
+- Research on a high-level variant of Cmm which abstracts away closure
+  representation and can be efficiently mapped to platforms providing host
+  garbage collection (e.g. wasm-gc, JavaScript, JVM). This will enable us to
+  avoid relying on a hand-written custom garbage collector and improve the
+  runtime reliability significantly.
```

Comment:
Though this is slightly more technical, I thought that reducing the (currently excessive) memory usage of the linker is another goal for Q3. Is it still in our list? Might also deserve to be mentioned in the "What may stop one from using Asterius right now" section, since it can be a problem for weaker machines.

Reply:
I added it to the "What may stop.." section. Performance optimization was planned in Q2 and I think all the low hanging fruits have been picked there